### PR TITLE
[15.0][IMP] stock_return_request: Improve performance and simplicity.

### PR DESCRIPTION
**Pull request by carlosdauden**
Avoid overload normal flows to cover a non usual flow.

@Tecnativa

### CODE REVIEW

**Comment by carlosdauden** [comment]
ping @chienandalu @sergio-teruel

**Comment by rousseldenis** [comment]
@carlosdauden As you remove a part of functionality, maybe you can a little bit explain why. Thanks

File: stock_return_request/models/stock_move.py [hunk 1]
```diff
@@ -30,9 +21,9 @@ def _compute_qty_returnable(self):
                 if move.state == "done":
                     move.qty_returnable = move.quantity_done
                 else:
+                    # TODO: Review if this option has sense
                     move.qty_returnable = move.reserved_availability
```
Comment (chienandalu) [inline stock_return_request/models/stock_move.py:25]: Maybe not so much. If there was a bussiness logic I can't remember anymore...

**Review by chienandalu — APPROVED** [review]
:+1: Agreed overall

**Comment by chienandalu** [comment]
> @carlosdauden As you remove a part of functionality, maybe you can a little bit explain why. Thanks

Well, the functionality is not being lost. The main purpose is to turn a computed stored field into a non stored one so we can release the load of its compute outside of its intended flows

File: stock_return_request/models/stock_return_request.py [hunk 2]
```diff
@@ -502,6 +501,7 @@ def _get_returnable_move_ids(self):
             moves = stock_move_obj.search(
                 line._get_moves_domain(), order=line.request_id.return_order
             )
+            moves = moves.filtered(lambda m: m.qty_returnable > 0.0)
```
Comment (rousseldenis) [inline stock_return_request/models/stock_return_request.py:504]: You should use float_compare()

Comment (carlosdauden) [reply stock_return_request/models/stock_return_request.py:504]: Changes done

File: stock_return_request/models/stock_move.py [hunk 3]
```diff
@@ -11,17 +11,8 @@ class StockMove(models.Model):
         string="Returnable Quantity",
         compute="_compute_qty_returnable",
         readonly=True,
-        store=True,
-        recursive=True,
     )
 
-    @api.depends(
```
Comment (pedrobaeza) [inline stock_return_request/models/stock_move.py:18]: **question:** Why removing `depends`? Although being non stored, the depends indicates when to mark the result as dirty if cached AFAIK.

**Review by pedrobaeza — COMMENTED** [review]
I have rebased and fix conflict, but there's one doubt inline.

**Review by pedrobaeza — APPROVED** [review]
Well, there's no answer on my question, but the optimization remains interesting, so let's merge it:

/ocabot merge patch

Please fw-port it to upper versions

**Pull request merged**